### PR TITLE
[React] Update React version to 18

### LINKED ISSUE #18416: Update React version to 18
##### **Overview of the feature request**

Keep react updated to latest version

##### **Related issues or PR**

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #18416 
---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/react/package.json [hunk 1]
```diff
@@ -27,19 +27,19 @@
...
+    "@types/react": "18.0.11",
     "@types/react-dom": "18.0.5",
     "@types/react-redux": "7.1.24",
     "@types/react-router-dom": "5.3.3",
     "@types/redux": "3.6.31",
     "@types/webpack-env": "1.17.0",
     "@typescript-eslint/eslint-plugin": "5.27.0",
     "@typescript-eslint/parser": "5.27.0",
-    "autoprefixer": "10.4.7",
+    "autoprefixer": "10.4.5",
```
Comment (mshima) [inline generators/client/templates/react/package.json:42]:
```suggestion
    "autoprefixer": "10.4.7",
```

**Review by mshima — APPROVED** [review]
LGTM

**Comment by qmonmert** [comment]
@mshima I had a warning with autoprefixer in version 10.4.7

**Comment by mshima** [comment]
@qmonmert a warning will not prevent dependabot auto update, so we should decide if the warning is ok or we need to take some action.
The warning happens with v17 too?

**Comment by qmonmert** [comment]
image
yes it happens also with react 17

**Comment by mshima** [comment]
Bootstrap 5.2.0 should fix it.
I don't think we can do anything right now.

**Comment by qmonmert** [comment]
ok 🙂

**Comment by qmonmert** [comment]
@mshima you merge?

**Pull request merged**